src/StepFactory.php

```diff
@@ -32,11 +32,21 @@ public static function create(array $step)
     /**
      * @param array $step
      *
+     * @return array
+     */
+    private static function getEntitiesFromStepData(array $step = [])
+    {
+        return array_key_exists('entities', $step) ? $step['entities'] : [];
```

Comment:
`array_key_exists` return `true` if `$step['entities']` is set to `NULL` which in this case this method will return `NULL` not an `array`. It's probably better to used `isset` instead (which return `false` in case of `NULL` value) to ensure that `getEntitiesFromStepData()` return an `array`. What do you think ?